pkg/build/apis/build/types.go

```diff
@@ -609,16 +609,33 @@ type BuildSource struct {
 	Secrets []SecretBuildSource
 }
 
-// ImageSource describes an image that is used as source for the build
+// ImageSource is used to describe build source that will be extracted from an image or used during a
+// multi stage build. A reference of type ImageStreamTag, ImageStreamImage or DockerImage may be used.
+// A pull secret can be specified to pull the image from an external registry or override the default
+// service account secret if pulling from the internal registry. Image sources can either be used to
+// extract content from an image and place it into the build context along with the repository source,
+// or used directly during a multi-stage Docker build to allow content to be copied without overwriting
+// the contents of the repository source (see the 'paths' and 'as' fields).
 type ImageSource struct {
-	// From is a reference to an ImageStreamTag, ImageStreamImage, or DockerImage to
+	// from is a reference to an ImageStreamTag, ImageStreamImage, or DockerImage to
 	// copy source from.
 	From kapi.ObjectReference
 
-	// Paths is a list of source and destination paths to copy from the image.
+	// A list of image names that this source will be used in place of during a multi-stage Docker image
+	// build. For instance, a Dockerfile that uses "COPY --from=nginx:latest" will first check for an image
+	// source that has "nginx:latest" in this field before attempting to pull directly. If the Dockerfile
+	// does not reference an image source it is ignored. This field and paths may both be set, in which case
+	// the contents will be used twice.
+	// +optional
+	As []string
+
+	// paths is a list of source and destination paths to copy from the image. This content will be copied
+	// into the build context prior to starting the build. If no paths are set, the build context will
+	// not be altered.
+	// +optional
```

Comment:
have you validated that the rest of the build logic properly handles this being empty?

Reply:
Went over all the uses of image paths - there weren't many.  We didn't actually ever require them to be set that I could find.